File: app-java/src/main/java/net/mEmoZz/yts/java/ui/splash/SplashContract.java
package net.mEmoZz.yts.java.ui.splash;

import net.mEmoZz.yts.java.ui.base.BaseInteractor;
import net.mEmoZz.yts.java.ui.base.BasePresenter;
import net.mEmoZz.yts.java.ui.base.BaseView;

/**
 * Authored by Mohamed Fathy on 10 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

public interface SplashContract {

  interface View extends BaseView<Presenter> {

    void openMainActivity();

    @Override default void setActionBar() {
    }

    @Override default void showProgressBar() {
    }

    @Override default void hideProgressBar() {
    }
  }

  interface Presenter extends BasePresenter<View, BaseInteractor> {
  }
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/splash/SplashContract.kt
package net.mEmoZz.yts.kotlin.ui.splash

/**
 * Authored by Mohamed Fathy on 10 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

interface SplashContract {

  interface View {

    fun setPresenter(presenter: SplashPresenter)

    fun openMainActivity()
  }

  interface Presenter {

    fun onAttach(view: SplashContract.View)

    fun onDestroy()
  }
}
